### DISCUSSION TITLE
Does the file upload to S3 also support Cloudfront?

### DISCUSSION THREAD

**Discussion by g1sbi** [pocketbase/pocketbase#2737] [Q&A] [answered]
I was planning on having a Cloudfront layer in front of my S3 bucket. Right now I'm using the direct S3 implementation provided by Pocketbase. Does it also support Cloudfront, or does that have to be implemented manually and forfeit using Pocketbase's file upload entirely?

**Comment by ganigeorgiev** [comment, marked as answer]
There is no builtin option for loading files from CDN endpoint. This was requested before, but while working on the implementation I've stumbled on several issues and couldn't found a reliable way that will work with every S3 provider. You can read more in https://github.com/pocketbase/pocketbase/issues/28#issuecomment-1492516269.

The recommended options for now are:

- Add `/api/files/*` static files cache on nginx/apache/etc. level

- Since you are using a CDN, your assets are most likely public accessible anyway so in this case instead of using the PocketBase file url, you can instead point directly to the CDN endpoint, which would be in the format:
    ```
    https://cdnurl.com/{collectionId}/{recordId}/{filename}
    ```
    The drawback is that thumbs can't be generated lazily this way.
